packages/runtime-core/src/renderer.ts

```diff
@@ -1327,7 +1332,7 @@ function baseCreateRenderer(
         }
         toggleRecurse(instance, true)
 
-        if (el && hydrateNode) {
+        if (el && hydrateNode && !isHmrReloading) {
```

Comment:
This is not the proper fix. if the initial VNode has `el` it means it needs hydration. so the proper fix should clear `vnode.el` before `reload`.

https://github.com/vuejs/core/blob/1bad606eb37903ae1ca430f2e4bc2f3e35c44ca8/packages/runtime-core/src/apiCreateApp.ts#L383-L392

changes to
```ts
const cloned = cloneVNode(vnode)
cloned.el = null
render( 
  cloned, 
  rootContainer, 
  namespace as ElementNamespace, 
) 
```